tree/ntuple/src/RField.cxx

```diff
@@ -830,8 +831,13 @@ void ROOT::RNullableField::ReconcileOnDiskField(const RNTupleDescriptor &desc)
    static const std::vector<std::string> prefixes = {"std::optional<", "std::unique_ptr<"};
 
    const auto &fieldDesc = desc.GetFieldDescriptor(GetOnDiskId());
-   EnsureMatchingOnDiskField(fieldDesc, kDiffTypeName);
-   EnsureMatchingTypePrefix(fieldDesc, prefixes);
+   try {
+      EnsureMatchingOnDiskField(fieldDesc, kDiffTypeName);
+      EnsureMatchingTypePrefix(fieldDesc, prefixes);
+   } catch (const RException &) {
+      fSubfields[0]->SetOnDiskId(GetOnDiskId());
```

Comment:
This looks a bit sketchy to me: since we're using RException for everything, how can we know that it's not a fatal exception, possibly coming from a deep call inside the Ensure methods?

Unrelated, for the Ensure methods I would adopt the same approach as e.g. `VerifyXxHash3`, where the call returns a `RResult` rather than throwing an exception directly (of course in a separate PR)

Reply:
> Unrelated, for the Ensure methods I would adopt the same approach as e.g. `VerifyXxHash3`, where the call returns a `RResult` rather than throwing an exception directly (of course in a separate PR)

Makes sense. See https://github.com/root-project/root/pull/19971